src/auto-evo/AutoEvoExploringTool.cs

```diff
@@ -0,0 +1,640 @@
+        useBiodiversityForceSplitCheckBox.Pressed = autoEvoConfiguration.UseBiodiversityForceSplit;
+
+        // Connect custom dropdown handler
+        historyListMenu.Popup.Connect("index_pressed", this, nameof(HistoryListMenuIndexChanged));
+        speciesListMenu.Popup.Connect("index_pressed", this, nameof(SpeciesListMenuIndexChanged));
+
+        // Mark ready, later on autoEvoConfiguration values should only be changed through GUI
+        ready = true;
+    }
+
+    public override void _Process(float delta)
+    {
+        base._Process(delta);
+
+        if (autoEvoRun != null)
+        {
+            runStatusLabel.Text = autoEvoRun.Status;
+
+            if (autoEvoRun.WasSuccessful)
+            {
+                ApplyAutoEvoRun();
+
+                // Clear autoEvoRun and enable buttons to allow the next run to start.
+                autoEvoRun = null;
+                finishOneGenerationButton.Disabled = false;
+                runOneStepButton.Disabled = false;
+                abortButton.Disabled = true;
+            }
+            else if (autoEvoRun.Aborted)
+            {
+                autoEvoRun = null;
+                finishOneGenerationButton.Disabled = false;
+                runOneStepButton.Disabled = false;
+                abortButton.Disabled = true;
+            }
+        }
+    }
+
+    [RunOnKeyDown("ui_cancel")]
+    public void AskExit()
+    {
+        exitConfirmationDialog.PopupCentered();
+    }
+
+    private void OnBackButtonPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        AskExit();
+    }
+
+    private void ConfirmExit()
+    {
+        TransitionManager.Instance.AddSequence(ScreenFade.FadeType.FadeOut, 0.1f,
+            SceneManager.Instance.ReturnToMenu, false);
+    }
+
+    private void ChangeTab(int index)
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        switch ((TabIndex)index)
+        {
+            case TabIndex.Config:
+            {
+                historyReportSplit.Visible = false;
+                configTab.Visible = true;
+                break;
+            }
+
+            case TabIndex.Report:
+            {
+                configTab.Visible = false;
+                viewerTab.Visible = false;
+                speciesSelectPanel.Visible = false;
+                historyReportSplit.Visible = true;
+                reportTab.Visible = true;
+                break;
+            }
+
+            case TabIndex.Viewer:
+            {
+                reportTab.Visible = false;
+                configTab.Visible = false;
+                speciesSelectPanel.Visible = true;
+                historyReportSplit.Visible = true;
+                viewerTab.Visible = true;
+                break;
+            }
+        }
+    }
+
+    /// <summary>
+    ///   This function updates all configurations in a row to avoid adding numerous separate callback functions.
```

Comment:
This is a bit lazy, but I'll allow it as this isn't a core gameplay component that should have the best quality code.

Reply:
This doesn't consume much performance but saves quite a bit code and in case more config will be there.

Reply:
Still other places in the game have bothered (for example the options menu) to set individual callbacks for controls, but as I said I'll allow it here but I wanted to say I don't consider this a very clean coding practice.